sql/core/src/test/scala/org/apache/spark/sql/UDFSuite.scala

```diff
@@ -151,6 +152,41 @@ class UDFSuite extends QueryTest with SharedSQLContext {
     assert(sql("SELECT strLenScala('test', 1)").head().getInt(0) === 5)
   }
 
+  test("Deterministic and literal UDF optimization") {
+    def udfNodesCount(plan: LogicalPlan): Int = {
```

Comment:
Might this `udfNodesCount` function be pulled up outside of this test for future reuse in other tests?

Reply:
Thanks @pattern for your comment. I was following the precedence in this test suite and since this function is only used for this particular test case, it did not seem necessary to pull it outside.  I don't have a strong preference either way..  so would like to keep it 'as is' if that is ok.